File: app/src/main/java/me/echeung/moemoekyun/client/socket/response/EventNotificationResponse.java
package me.echeung.moemoekyun.client.socket.response;

import lombok.Getter;
import me.echeung.moemoekyun.client.model.Event;

@Getter
public class EventNotificationResponse extends NotificationResponse {
    public static final String TYPE = "EVENT";

    private Details d;

    @Getter
    public static class Details extends NotificationResponse.Details {
        private Event event;
    }
}

File: app/src/main/kotlin/me/echeung/moemoekyun/client/api/socket/response/EventNotificationResponse.kt
package me.echeung.moemoekyun.client.api.socket.response

import kotlinx.serialization.Serializable
import me.echeung.moemoekyun.client.api.model.Event

@Serializable
class EventNotificationResponse : BaseResponse() {
    val t: String? = null
    val d: Details? = null

    @Serializable
    data class Details(
        val type: String,
        val event: Event
    )

    companion object {
        const val TYPE = "EVENT"
    }
}
